packages/block/src/header.ts

```diff
@@ -300,9 +298,13 @@ export class BlockHeader {
         throw new Error(msg)
       }
       const londonHfBlock = this._common.hardforkBlock(Hardfork.London)
-      if (isTruthy(londonHfBlock) && this.number === londonHfBlock) {
+      if (
+        typeof londonHfBlock === 'bigint' &&
+        londonHfBlock !== BigInt(0) &&
```

Comment:
Yeah, I think it should be fine (not sure if any of our previous tests relied on BigInt(0) falsiness though, we'll see with the CI). Removed that check!

Reply:
Actually a couple of tests fail if I remove those checks, so I will revert back.

Reply:
Just to note: having no additional tests fail is *not* a sufficient indicator that things are correct, there is absolutely no guarantee that some possible error case is covered by any test. So we should always do these kind of decisions "by reasoning", otherwise this is too dangerous.

Also to confirm: yes, HF on block 0 is an important use case.